src/plugins/workspace/public/components/global_search/search_pages_command.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ChromeNavLink, ChromeRegistrationNavLink, CoreStart } from 'opensearch-dashboards/public';
+import { first } from 'rxjs/operators';
+import React, { ReactNode } from 'react';
+import { BehaviorSubject } from 'rxjs';
+import { getFirstUseCaseOfFeatureConfigs } from '../../utils';
+import { GlobalSearchPageItem } from './page_item';
+import { DEFAULT_NAV_GROUPS, NavGroupType } from '../../../../../core/public';
+import { WorkspaceUseCase } from '../../types';
+
+export const searchPageWithInWorkspace = async (
+  query: string,
+  registeredUseCases$: BehaviorSubject<WorkspaceUseCase[]>,
+  coreStart?: CoreStart,
+  callback?: () => void
+): Promise<ReactNode[]> => {
+  if (coreStart) {
+    const currentWorkspace = await coreStart.workspaces.currentWorkspace$.pipe(first()).toPromise();
+
+    const navGroupMap = await coreStart.chrome.navGroup
+      .getNavGroupsMap$()
+      .pipe(first())
+      .toPromise();
+
+    const allAvailableCaseId = Object.entries(DEFAULT_NAV_GROUPS)
```

Comment:
So these are global nav item, it will be searchable even when outside of a workspace, if that's the case, is the function name `searchPageWithInWorkspace` still appropriate?

Reply:
make sense, update the method name to `searchPages` so that it will not limit to pages in workspace.